src/deluge/gui/ui/keyboard/layout/norns.cpp

```diff
@@ -47,9 +47,9 @@ void KeyboardLayoutNorns::renderPads(RGB image[][kDisplayWidth + kSideBarWidth])
 		for (int32_t x = 0; x < kDisplayWidth; x++) {
 			int32_t note = noteFromCoords(x, y);
 
-			// If highlighting notes is active, do it
-			if (getHighlightedNotes()[note] != 0) {
-				image[y][x] = colours::white_full.adjust(getHighlightedNotes()[note], 1);
+			// If norns is active, do it
+			if (getNornsNotes()[note] != 0) {
+				image[y][x] = colours::white_full.adjust(getNornsNotes()[note], 1);
```

Comment:
does this break regular note highlighting?

Reply:
It shouldn't. Previously, the incoming midi notes from midigrid intended for updating the pad LEDs in norns view would be saved to the highlightedNotes array. If you opened a clip with another keyboard layout, it'd render all the noise from midigrid. Or if you had incoming midi notes, it'd interfere with the norns view and highlight things that shouldn't be lit up. I tested it a bit last night where I had an incoming midi sequence on channel 1 highlighting on isomorphic and the norns view highlighting from midigrid stream.